### DISCUSSION TITLE
Doubts on operation

### DISCUSSION THREAD

**Discussion by ldelvalleh** [PricelessToolkit/ESPBell-MAX#5] [Q&A]
Hello

I purchased the ESPBell-MAX and have two questions to ask. The first is if I can open the door with some control entity from Home Assistant since by default the entity does not appear. I have only seen what can be done with notification.

And the second is if the Arduino code is ported to ESPHome.

Thank you.

**Comment by PricelessToolkit** [comment]
For opening, you need to publish an MQTT command with automation.
![mqtt_command](https://github.com/PricelessToolkit/ESPBell-MAX/assets/81021972/38536552-a297-414f-9bf1-237a20cffb7d)

**Reply by ldelvalleh** [reply]
Hello

Thanks for answering. Yes, I understand, my doubt was that the ESPBell-Max can somehow be woken up to open the door but I think not, since the sleeping state is done by hardware and while it is asleep it cannot receive MQTT messages. By the way, a great idea to save a lot of energy.

I have already bought an ESPBell Lite because this can be done as I have seen. I'll try it.

Thank you so much.

**Comment by PricelessToolkit** [comment]
ESPBell-Max only Arduino, Lite version works with ESPHome

**Reply by ldelvalleh** [reply]
Ok, I'm going to try to migrate the code from Arduino to ESPHome, do you think it can be done? If I can do it I'll pass it on to you.

By the way, a suggestion/question, wouldn't there be possibilities of putting some pins for connecting the pins to load the code? Pogo pins are fine, but they are another thing to buy and soldering is easy for some, but there are many other people who don't dare.

Not having this facility to upload a code limits the number of people who have access to this wonderful device.

It's just a suggestion.

**Reply by PricelessToolkit** [reply]
I haven't even tried to write code for ESPBell-MAX on ESPHome. First thing you need to do is pull up one pin after boot, when I get home I'll write pin number.

**Reply by PricelessToolkit** [reply]
better late than never :( sorry
The first thing you need to do is pull  GPIO4 PIN HIGH, then it will stay on, when your code finished doing its own thing, pull GPIO4 LOW, and it will turn off.
